sdk/plugin_test.go

```diff
@@ -844,3 +845,22 @@ func TestListenSettings_Validate_Error(t *testing.T) {
 		assert.Equal(t, testCase.errCount, len(merr.Errors), merr.Error())
 	}
 }
+
+// TestRedactedJsonConfigOutput tests that we redact passwords when we dump
+// them to the logs. map[string]interface{} values will be replaced with
+// REDACTED if the lowercase map key containes "pass".
+func TestRedactedJsonConfigOutput(t *testing.T) {
+	// Test string from a log before we redacted passwords.
+	s := "{\"Version\":\"1\",\"Debug\":true,\"Settings\":{\"Mode\":\"serial\",\"Listen\":{\"Enabled\":true,\"Buffer\":100},\"Read\":{\"Enabled\":true,\"Interval\":\"3s\",\"Buffer\":1024},\"Write\":{\"Enabled\":true,\"Interval\":\"1s\",\"Buffer\":1024,\"Max\":100},\"Transaction\":{\"TTL\":\"5m\"},\"Cache\":{\"Enabled\":false,\"TTL\":180000000000}},\"Network\":{\"Type\":\"tcp\",\"Address\":\"0.0.0.0:5003\",\"TLS\":null},\"DynamicRegistration\":{\"Config\":[{\"authenticationPassphrase\":\"Vap0rIsDriven!\",\"authenticationProtocol\":\"SHA\",\"endpoint\":\"10.193.3.201\",\"model\":\"PXGMS UPS + EATON 93PM\",\"port\":161,\"privacyPassphrase\":\"Vap0rIsDriven!\",\"privacyProtocol\":\"AES\",\"userName\":\"user\",\"version\":\"v3\"}]},\"Limiter\":null,\"Health\":{\"UseDefaults\":true},\"Context\":{}}"
+
+	expected := `{"Context":{},"Debug":true,"DynamicRegistration":{"Config":[{"authenticationPassphrase":"REDACTED","authenticationProtocol":"SHA","endpoint":"10.193.3.201","model":"PXGMS UPS + EATON 93PM","port":161,"privacyPassphrase":"REDACTED","privacyProtocol":"AES","userName":"user","version":"v3"}]},"Health":{"UseDefaults":true},"Limiter":null,"Network":{"Address":"0.0.0.0:5003","TLS":null,"Type":"tcp"},"Settings":{"Cache":{"Enabled":false,"TTL":180000000000},"Listen":{"Buffer":100,"Enabled":true},"Mode":"serial","Read":{"Buffer":1024,"Enabled":true,"Interval":"3s"},"Transaction":{"TTL":"5m"},"Write":{"Buffer":1024,"Enabled":true,"Interval":"1s","Max":100}},"Version":"1"}`
+
+	output, err := RedactPasswords(s)
+	if err != nil {
+		fmt.Printf("Error: %v\n", err)
+	}
+
+	if output != expected {
+		t.Errorf("actual != expected,  expected: %v, actual: %v", expected, output)
+	}
```

Comment:
For consistency with the other tests in the SDK code base, can we use the `assert` package for this?
```go
assert.NoError(t, err)
assert.Equal(t, expected, output)
```

Reply:
Done